addons/binding/org.openhab.binding.homie/src/main/java/org/openhab/binding/homie/handler/HomieDeviceHandler.java

```diff
@@ -0,0 +1,421 @@
+            } else if (topic.isESHStep()) {
+                this.step = new BigDecimal(message);
+                updateChannelType();
+            } else if (topic.isESHItemType()) {
+                this.itemType = message;
+                updateChannelType();
+            } else if (topic.isESHDescription()) {
+                this.description = message;
+                updateChannelType();
+            } else if (topic.isESHValue()) {
+                State state = transformToState(message);
+                updateState(getChannelUID(), state);
+            }
+
+        }
+
+        private State transformToState(String message) {
+            State result = null;
+            if (StringUtils.equals(itemType, "Switch")) {
+                boolean on = StringUtils.equalsAnyIgnoreCase(message, "1", "ON", "TRUE");
+                result = on ? OnOffType.ON : OnOffType.OFF;
+            } else if (StringUtils.equals(itemType, "Number")) {
+                result = new DecimalType(message);
+            } else if (StringUtils.equals(itemType, "Rollershutter")) {
+                boolean up = StringUtils.equalsAnyIgnoreCase(message, "1", "ON", "TRUE", "UP");
+                result = up ? UpDownType.UP : UpDownType.DOWN;
+            } else if (StringUtils.equals(itemType, "Color")) {
+                result = new HSBType(message);
+            } else if (StringUtils.equals(itemType, "Contact")) {
+                boolean open = StringUtils.equalsAnyIgnoreCase(message, "1", "ON", "TRUE", "OPEN");
+                result = open ? OpenClosedType.OPEN : OpenClosedType.CLOSED;
+            } else if (StringUtils.equals(itemType, "Dimmer")) {
+                result = new PercentType(message);
+            } else if (StringUtils.equals(itemType, "Player")) {
+                boolean play = StringUtils.equalsAnyIgnoreCase(message, "1", "ON", "TRUE", "PLAY");
+                result = play ? PlayPauseType.PLAY : PlayPauseType.PAUSE;
+            } else if (StringUtils.equals(itemType, "String")) {
+                result = new StringType(message);
+            } else {
+                logger.warn("Cannot transform message '{}' to a ESH state, using itemtype 'String' instead");
+                result = new StringType(message);
+            }
+            return result;
+        }
+
+    }
+
+    private class DefaultNodeHandler extends NodeHandler {
+
+        public DefaultNodeHandler(String nodeId, HomieConfiguration config) throws MqttException {
+            super(nodeId, config, "");
+
+        }
+
+        @Override
+        protected void onMqttMessage(HomieTopic topic, String message) {
+            logger.warn("Handling non ESH Homie Nodes is not implemented yet");
+
+        }
+
+        @Override
+        protected void onHandleCommand(Command command) {
+            logger.warn("Handling non ESH Homie Nodes is not implemented yet");
+
+        }
+
+    }
+
+    private Map<String, NodeHandler> nodeHandlers = new HashMap<>();
```

Comment:
These handler classes unfortunately cannot be extracted to a dedicated file. They use methods of the parent class that are protected.